src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -113,6 +112,7 @@ function _getCacheFilePath(config) {
 
 function Loader(config, environment, resourceMap) {
   this._config = config;
+  this._CoverageCollector = require(config.coverageCollector);
```

Comment:
Sorry, the error was only because requiring ``undefined``. And why it worked when moving it bellow because the test didn't go into ``shouldCollectCoverage`` block. But I feel it makes some sense, if you don't need to collect coverage then you don't need require collector.